### ISSUE TITLE
Gentoo Linux logo is not present, like Arch Linux logo is on Arch.

### ISSUE THREAD

**Issue by Soccera1** [hyprwm/hyprsysteminfo#1]
Expected outcome: logo for Linux distribution used should appear, like in this screenshot from hyprwm on X.

![GZyFC0PXUAgCdNc](https://github.com/user-attachments/assets/574ddd9b-9c3a-4b30-97b8-f861b8af926f)

However, I have this screenshot which shows no logo on Gentoo Linux.

![2024-10-14-041148_hyprshot](https://github.com/user-attachments/assets/9bea472b-88f2-49dc-b179-98204c412b11)

Environment:

Gentoo Linux with default/linux/amd64/23.0/desktop/gnome/systemd (stable) profile
Hyprland 43.0
GCC
Latest hyprsysteminfo version

Steps to reproduce:

Install Gentoo Linux
Use GCC, cmake, and make to make and install hyprsysteminfo
Run hyprsysteminfo in Hyprland
Observe lack of Gentoo Linux logo

**Comment by vaxerski** [comment]
what's the output of `cat /etc/os-release`?

**Comment by Soccera1** [comment]
NAME=Gentoo
ID=gentoo
PRETTY_NAME="Gentoo Linux"
ANSI_COLOR="1;32"
HOME_URL="https://www.gentoo.org/"
SUPPORT_URL="https://www.gentoo.org/support/"
BUG_REPORT_URL="https://bugs.gentoo.org/"
VERSION_ID="2.15"

**Comment by vaxerski** [comment]
well that's your answer. Logo depends on this having a logo:

![image](https://github.com/user-attachments/assets/40c02580-c0c3-4303-b8a1-f0deb885d8e0)

**Comment by vaxerski** [comment]
maybe it's possible to fall back on id, gimme sec

**Comment by vaxerski** [comment]
hm. I am looking at my icons and I can't find a gentoo icon within my theme. Dunno how that should be obtained.

**Comment by Soccera1** [comment]
It might not be what you want for the project but one possible solution would be to have system that automatically downloads the correct logo if it's not present after detecting what distro it is. For example, it would download https://www.gentoo.org/assets/img/logo/gentoo-signet.svg if the detected distro is gentoo and there isn't a logo already.

**Comment by vaxerski** [comment]
No, allowing a utility to connect to the internet is not what I want it to do. Embedding icons would be passable, but idk what the policies of the gentoo logo are

**Comment by Soccera1** [comment]
I'm not a lawyer, but I'm pretty sure you'd be able to embed the icon for Gentoo. Here's the source: https://www.gentoo.org/inside-gentoo/foundation/name-logo-guidelines.html. I'm not sure about any other distros specifically, but most distros are OK with non commercial use as long as it's clearly not affiliated with their distro as far as I know.

**Comment by alba4k** [comment]
I guess you could also fall back to using nerd fonts for distros that don't provide logos

Or just embed it I guess, would make the whole thing a bit heavier but who cares, it's just 20 KB or so

**Comment by outfoxxed** [comment]
> hm. I am looking at my icons and I can't find a gentoo icon within my theme. Dunno how that should be obtained.

are you sure its not a special icon provided per distro? I wouldnt expect to have archlinux-logo on nix or nix-snowflake on arch.

**Comment by thesunexpress** [comment]
This is quite trivial to achieve. FreeBSD doesn't do any of that LOGO nonsense in its os-release file, but can be made to do so. Depending on what peculiar things your particular Linux distro does, the following may help.
Us FreeBSD folks rely on a script that generates `/etc/os-release` at boot time. Hence we do not actually touch this file directly & instead use `/etc/rc.d/os-release`, modifying it slightly by adding the line `LOGO="some-slick-name" ` before the __EOF.
In my case, vanilla will do, so I added `LOGO="freebsd" ` to my `/etc/rc.d/os-release` file -- which references a file named `freebsd.svg` , noting the spelling/caps & lack of `.svg` file extension in the actual `/etc/os-release` file. I'm not 100% sure how Linux goes about its os-release file, whether it is routinely regenerated or not, so your mileage may vary on how to keep a `LOGO="some-name" ` defined in your os-release file. Modifications to FreeBSD RC.d scripts survive OS update/rebuild, if you know what you are doing, but easy enough to re-do should you forget.
Go search on the internet for a nice logo you'd like to use for your flavor of Linux. Probably the distro's primary website might a good start. I went with a common FreeBSD SVG -- note: using an SVG is probably the best course of action here, for obvious reasons. Now copy that `some-slick-name.svg` file you found on the internet, to system-wide shared icons directory. For FreeBSD that happens to be `/usr/local/share/icons/hicolor/scalable/apps/`. Assuming your Linux distro is somewhat sane, a similar system-wide shared icons directory probably lives in `/usr/share/icons/hicolor/` -- adjust as needed. The `scalable` subdirectory is the best place for SVGs, the `apps` subdirectory within, being the best place for icons called-up by apps. Remember to omit the file extension in the os-release file; it may / may not be required for Linux... Unsure if you need to reboot a Linux box for the `/etc/os-release` file to be properly sourced, easy enough to find out.
Once you've set all the things, voila! A `hyprsysteminfo` with fancy OS branding logo.
(Yes, I massaged the `src/main.qml` definition file a bit to suit my needs/OCD better)

![Image](https://github.com/user-attachments/assets/aac967a8-b18e-4815-bbc1-e2fda98f5d34)
